src/unix/thread.c

```diff
@@ -72,10 +75,59 @@ int uv_thread_create(uv_thread_t *tid, void (*entry)(void *arg), void *arg) {
 }
 
 
+int uv_thread_create_on(int proc_num,
+                        uv_thread_t *tid,
+                        void (*entry)(void *arg),
+                        void *arg) {
```

Comment:
Thanks for the pointers to other OS ways of managing this. Sheesh, what a mess!

I have no problem with your proposed API. I'd suggest a small change though--adding features, or rather attributes, might be more naturally done as:

```
typedef struct uv_thread_attr_s uv_thread_attr_t;

enum uv_thread_attr_actions {
  UV_THREAD_ATTR_GET_AFFINITY,
  UV_THREAD_ATTR_SET_AFFINITY
  /* Macro for feature detection. */
#define UV_THREAD_ATTR_GET_AFFINITY UV_THREAD_ATTR_GET_AFFINITY
#define UV_THREAD_ATTR_SET_AFFINITY UV_THREAD_ATTR_SET_AFFINITY
};

struct uv_thread_affinity_t {
  char* cpumask;
  size_t cpumask_size;  /* in bytes */
};

struct uv_thread_attr_s {
  int action;
  union {
    struct uv_thread_affinity_t affinity;
  } data;
};
```

This can go one level below the versioning scheme, if you think that's still necessary. Also, it would be ideal to have a chain of ``uv_thread_attr_t``s, so that I can set affinity, detached state, and other attributes in a single call, as pthreads does it. What's the preferred way to do this?

This would work fine for ``uv_thread_set_attr()`` and ``uv_thread_get_attr()`` as well.

Any comment on ``uv_thread_detach()``?